Lesson-3/assignment/Payroll.sol

```
@@ -0,0 +1,88 @@
+pragma solidity ^0.4.18;
+
+import './SafeMath.sol';
+import './Ownable.sol';
+
+contract Payroll is Ownable{
+    using SafeMath for uint;
+    struct Employee {
+        address id;
+        uint salary;
+        uint lastPayday;
+    }
+    uint constant payDuration = 10 seconds;
+    uint private totalSalary = 0;
+    address owner;
+    mapping(address => Employee) public employees;
+    
+    modifier employeeExists(address employeeId){
+        var employee = employees[employeeId];
+        require(employee.id != 0x0);
+        _;
+    }
+    
+    modifier deleteEmployee(address employeeId){
+        _;
+        delete employees[employeeId];
+    }
+    
+    function _partialPaid(Employee employee) private {
+        uint payment = employee.salary.mul(now.sub(employee.lastPayday)).div(payDuration);
+        _payEmployee(employee, payment, now);
+    }
+    
+    function _payEmployee(Employee employee, uint payment, uint time) private 
+    {
+        employee.lastPayday = time;
+        employee.id.transfer(payment);
+    }
+    
+    function addEmployee(address employeeId, uint salary) public onlyOwner{
+        var employee = employees[employeeId];
+        if (employee.id == 0x0) {
```

silently fail不是很好，这个地方可以用require，更好的做法是用employeeNotExist(employeeId)